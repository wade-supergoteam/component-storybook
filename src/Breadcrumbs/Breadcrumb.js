import PropTypes from "prop-types";
import styled from "styled-components";
import { borderRadius } from "../utils/border-radius";
import themeProp from "../utils/theme";
import {
  breadcrumbActiveColor,
  breadcrumbBg,
  breadcrumbDivider,
  breadcrumbDividerColor,
  breadcrumbItemPadding,
  breadcrumbMarginBottom,
  breadcrumbPaddingX,
  breadcrumbPaddingY,
} from "./default-theme";

const Breadcrumb = styled.ol`
  display: flex;
  flex-wrap: wrap;
  padding: ${themeProp("breadcrumbPaddingY", breadcrumbPaddingY)} ${themeProp("breadcrumbPaddingX", breadcrumbPaddingX)};
  margin-bottom: ${themeProp("breadcrumbMarginBottom", breadcrumbMarginBottom)};
  list-style: none;
  background-color: ${themeProp("breadcrumbBg", breadcrumbBg)};
  ${borderRadius(themeProp("bordeRadius", ".25rem"))};
`;

Breadcrumb.Item = styled.li.attrs(props => ({
  "aria-current": props.active && "page",
}))`
  :not(:first-child)::before {
    display: inline-block;
    padding-right: ${themeProp("breadcrumbItemPadding", breadcrumbItemPadding)};
    padding-left: ${themeProp("breadcrumbItemPadding", breadcrumbItemPadding)};
    color: ${themeProp("breadcrumbDividerColor", breadcrumbDividerColor)};
    content: ${themeProp("breadcrumbDivider", breadcrumbDivider)};
  }

  :not(:first-child)&:hover::before {
    text-decoration: underline;
  }
  :not(:first-child)&:hover::before {
    text-decoration: none;
  }

  ${({ active }) => active && `color: ${themeProp("readcrumbActiveColor", breadcrumbActiveColor)};`};
`;

Breadcrumb.defaultProps = {};

Breadcrumb.Item.defaultProps = {
  active: false,
};

Breadcrumb.Item.Proptypes = {
  active: PropTypes.bool,
};

export default Breadcrumb;
