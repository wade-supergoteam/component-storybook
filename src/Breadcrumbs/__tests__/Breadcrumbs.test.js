import React from "react";
import renderer from "react-test-renderer";
import Breadcrumb from "../";

test("Breadcrums is created", () => {
    const wrapper = renderer.create(<Breadcrumb />);
    expect(wrapper).toMatchSnapshot();
});

test("Breadcrums item", () => {
    const wrapper = renderer.create(<Breadcrumb.Item active />);
    expect(wrapper).toMatchSnapshot();
});
