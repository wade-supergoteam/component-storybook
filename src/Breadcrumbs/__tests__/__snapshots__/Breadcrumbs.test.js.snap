// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrums is created 1`] = `
<ol
  className="sc-bdVaJa flmxlQ"
/>
`;

exports[`Breadcrums item 1`] = `
<li
  aria-current="page"
  className="sc-bwzfXH cdQKDC"
/>
`;

exports[`breadcrums is created 1`] = `
<ol
  className="sc-bdVaJa flmxlQ"
/>
`;

exports[`create li element 1`] = `
<li
  aria-current={false}
  className="sc-bwzfXH jjlExb"
/>
`;
